src/bin/krimedc.rs

```diff
@@ -496,14 +510,25 @@ async fn keytab_extract_run(name: String, output: PathBuf, config: Config) -> io
         Name::principal(name.as_str(), server_state.realm.as_str())
     };
 
-    let principal_record = server_state
-        .principals
-        .get(&principal_name)
-        .ok_or_else(|| {
-            std::io::Error::new(std::io::ErrorKind::InvalidInput, "no matching principal")
-        })?;
+    let key: DerivedKey = if principal_name.is_service_krbtgt(&server_state.realm) {
```

Comment:
Why do we want to extract the primary key here?

Reply:
To decrypt the tickets in wireshark

Reply:
But we can make it available only in debug builds